File: analyst-ui/services/charterPlayground/charterPlayground.schema.ts
import { z } from "zod";

export const CharterPlaygroundUpdateSchema = z.object({
  query: z.string().optional(),
  name: z.string().optional(),
});

export const CharterPlaygroundResponseSchema = z.object({
  charter_id: z.number(),
  user_id: z.number(),
  public_id: z.string(),
  name: z.string(),
  query: z.string(),
  created_at: z.string(),
  updated_at: z.string(),
});

export const CharterPlaygroundListResponseSchema = z.array(CharterPlaygroundResponseSchema);

export type CharterPlaygroundUpdate = z.infer<typeof CharterPlaygroundUpdateSchema>;
export type CharterPlaygroundResponse = z.infer<typeof CharterPlaygroundResponseSchema>;
export type CharterPlaygroundListResponseItem = z.infer<typeof CharterPlaygroundResponseSchema>;
export type CharterPlaygroundListResponse = z.infer<typeof CharterPlaygroundListResponseSchema>;

// Query result schema
export const CharterPlaygroundQueryResultSchema = z.object({
  data: z.array(z.record(z.string(), z.any())).optional().nullable(),
  error: z.string().optional().nullable(),
});

export type CharterPlaygroundQueryResult = z.infer<typeof CharterPlaygroundQueryResultSchema>;

File: ai-backend/src/services/charter_playground/charter_playground_schema.py
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel


# Charter Playground CRUD operations schema
class CharterPlaygroundCreate(BaseModel):
    charter_id: int
    user_id: int


class CharterPlaygroundUpdate(BaseModel):
    query: Optional[str]
    name: Optional[str]


class CharterPlaygroundResponse(BaseModel):
    charter_id: int
    user_id: int
    public_id: str
    name: str
    query: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class ExecuteCharterPlaygroundQuery(BaseModel):
    query: str
    limit: Optional[int] = None


class ExecuteCharterPlaygroundResponse(BaseModel):
    data: Optional[List[dict]]
    error: Optional[str]
